server/src/main/java/org/elasticsearch/search/profile/query/ProfileCollectorManager.java

```diff
@@ -13,73 +13,31 @@
 import org.apache.lucene.search.CollectorManager;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 
 /**
  * A {@link CollectorManager} that takes another CollectorManager as input and wraps all Collectors generated by it
  * in an {@link InternalProfileCollector}. It delegates all the profiling to the generated collectors via {@link #getCollectorTree()}
  * and joins the different collector trees together when its {@link #reduce} method is called.
- * Supports optionally providing sub-collector managers for top docs as well as aggs collection, so that each
- * {@link InternalProfileCollector} created is provided with the corresponding sub-collectors that are children of the top-level collector.
+ * Note: does not support children profile collectors.
  * @param <T> the return type of the wrapped collector manager, which the reduce method returns.
  */
 public final class ProfileCollectorManager<T> implements CollectorManager<InternalProfileCollector, T> {
-
     private final CollectorManager<? extends Collector, T> collectorManager;
     private final String reason;
-    private final ProfileCollectorManager<?> topDocsSubCollectorManager;
-    private final ProfileCollectorManager<?> aggsSubCollectorManager;
-    // this is a bit of a hack: it allows us to retrieve the last collector that newCollector has returned for sub-collector managers,
-    // so that we can provide them to InternalProfileCollector's constructor as children. This is fine as newCollector does not get called
-    // concurrently, but rather in advance before parallelizing the collection
-    private InternalProfileCollector profileCollector;
```

Comment:
query phase is the only place where we have a top level collector wrapped with a profile collector, as well as its inner collectors (for top docs and aggs) both wrapped with a profile collector. Thanks to removing collector manager wrapping, this is now directly handled in QueryPhaseCollectorManager within newCollector, and this complicated bit can be removed from ProfileCollectorManager, which is only used by the dfs phase. There we do need to wrap the collector manager because we rely on the lucene top docs collector managers.